### PR TITLE
SDK-1558 Revoke Token

### PR DESCRIPTION
**Pull request by kleewho**
feat: Add revoke token feature

### CODE REVIEW

File: src/main/java/com/pubnub/api/endpoints/access/RevokeToken.java [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+        return PNOperationType.PNAccessManagerRevokeToken;
+    }
+
+    @Override
+    protected boolean isAuthRequired() {
+        return false;
+    }
+
+    private String repairEncoding(String token) {
+        String[] parts = token.split(" ");
```
Comment (budgetpreneur) [inline src/main/java/com/pubnub/api/endpoints/access/RevokeToken.java:98]: We never had a case where token comes with spaces. Any reason for splitting token with space?

File: src/test/java/com/pubnub/api/endpoints/access/GrantTokenEndpointTest.java [hunk 2]
```diff
@@ -27,6 +29,27 @@ public void beforeEach() throws IOException {
         pubnub.getConfiguration().setSecretKey("secretKey").setIncludeInstanceIdentifier(true);
     }
 
+    @Test
+    public void test() throws PubNubException {
+        PNConfiguration conf = new PNConfiguration();
+        conf.setPublishKey("pub-c-395744cc-604e-410e-9074-ade176d5aac0")
+                .setSubscribeKey("sub-c-fdc9b0fc-3e1a-11ea-9e28-4eda21f1b082")
+                .setSecretKey("sec-c-M2YzM2MxMDgtM2EyNS00ZTkxLWEzM2ItM2E5MTJlYTc0NGFi")
+                .setOrigin("ingress-tcp-pub.az1.pdx1.aws.int.ps.pn")
```
Comment (budgetpreneur) [inline src/test/java/com/pubnub/api/endpoints/access/GrantTokenEndpointTest.java:38]: hardcoded keys and origin.  Perhaps better to replace them with environment variable names or something else.

Comment (kleewho) [reply src/test/java/com/pubnub/api/endpoints/access/GrantTokenEndpointTest.java:38]: yeah. That shouldn't be here. Thanks

File: src/main/java/com/pubnub/api/endpoints/access/RevokeToken.java (hunk 1 above)
Comment (kleewho) [reply src/main/java/com/pubnub/api/endpoints/access/RevokeToken.java:98]: Yes. A test scenario in contract tests. That's why I disagree with the approach. I had to jump over some hoops just to satisfy the test

Comment (kleewho) [reply src/main/java/com/pubnub/api/endpoints/access/RevokeToken.java:98]: https://github.com/pubnub/sdk-specifications/blob/62e194d895dfac0e970dc2f6d9f3f76815de5b92/features/access/revoke-token.feature#L43

Comment (budgetpreneur) [reply src/main/java/com/pubnub/api/endpoints/access/RevokeToken.java:96]: Hm..I am surprised. I thought existing SDK has already logic of encoding chars. 
For example; When we get space in channel names or UUID, we encode it. So this logic already exist. Why not reuse the same logic? Is SDK encoding separately for each operation?

Comment (kleewho) [reply src/main/java/com/pubnub/api/endpoints/access/RevokeToken.java:96]: Well, we have it working for channels and most other parameters but it's done by okhttp which do not encodes everything that we have in the test case. Specifically it's not encoding the `=`. If I encode it manually and pass it to okhttp with `%20` okhttp will encode the `%` once again. So I had to disable the encoding for `token` done by okhttp and then encode it myself. Unfortunately the encoder used encodes `space` into `+` and not `%20`

**Review by budgetpreneur — APPROVED** [review]
LGTM

**Comment by parfeon** [comment]
If it will be released, don't forget to add changes in first comment.

**Comment by kleewho** [comment]
@client-engineering-bot release

File: src/test/java/com/pubnub/contract/access/step/GivenSteps.kt [hunk 3]
```diff
@@ -72,33 +75,82 @@ class GivenSteps(private val grantTokenState: GrantTokenState) {
         grant_permission(permissionType)
     }
 
+    //DELETE http://localhost:8090/v3/pam/sub-c-067a9e98-fb50-11eb-bc10-56aa1f324334/grant/A8TlaF6egF9dE6a0DzVpnEyftEt7Nl?pnsdk=PubNub-Kotlin%2F6.2.0&requestid=6cc73a57-d282-4685-bef1-7448895b8821&uuid=pn-6ecefeb0-4625-4dc2-b2da-ffea6c784678
```
Comment (bartk) [inline src/test/java/com/pubnub/contract/access/step/GivenSteps.kt:78]: @kleewho some keys leaking here in comments

**Review by bartk — APPROVED** [review]

**Review by bartk — DISMISSED** [review]

File: README.md [hunk 4]
```diff
@@ -28,7 +28,7 @@ You will need the publish and subscribe keys to authenticate your app. Get your
 
    * for Gradle, add the following dependency in your `gradle.build`:
      ```groovy
-     compile group: 'com.pubnub', name: 'pubnub-gson', version: '5.2.1'
+     compile group: 'com.pubnub', name: 'pubnub-gson', version: '5.2.3'
```
[the remark this answers is not in the document]
Comment (kleewho) [reply README.md:31]: It's not that important. It's going to be changed with the release.

File: src/test/java/com/pubnub/contract/access/step/GivenSteps.kt (hunk 3 above)
Comment (kleewho) [reply src/test/java/com/pubnub/contract/access/step/GivenSteps.kt:78]: I'll remove them

**Review by bartk — DISMISSED** [review]

**Review by techwritermat — DISMISSED** [review]

**Comment by kleewho** [comment]
@client-engineering-bot release

**Review by kazydek — APPROVED** [review]

**Review by bartk — APPROVED** [review]

**Pull request merged**